### PR TITLE
Upload: use graph objects

### PR DESCRIPTION
**Pull request by dorschw**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6103
fixes: [CIAC-5356](https://jira-hq.paloaltonetworks.local/browse/CIAC-5356)

## Description
Use the new upload mechanism instead of the ObjectFactory, allowing to use Preparer objects in the sdk-upload-command flow

[SDK Nightly build](https://code.pan.run/xsoar/content/-/pipelines/5215872)

### CODE REVIEW

File: demisto_sdk/commands/content_graph/parsers/mapper.py [hunk 1]
```diff
@@ -59,7 +59,9 @@ def connect_to_dependencies(self) -> None:
                 default_incident_type, content_type_to_map, is_mandatory=False
             )
 
-        for incident_type, mapping_data in self.json_data.get("mapping", {}).items():
+        for incident_type, mapping_data in (
+            self.json_data.get("mapping") or {}
```
Comment (dantavori) [inline demisto_sdk/commands/content_graph/parsers/mapper.py:63]: was this auto-suggested (and why)? asking out of curiosity

**Review by dantavori — COMMENTED** [review]
nice!

File: demisto_sdk/commands/content_graph/objects/pack.py [hunk 2]
```diff
@@ -63,6 +72,35 @@
...
+    Used to upload an existing zip file
+    """
+    if target_demisto_version < Version("6.5.0"):
+        raise RuntimeError(
+            "Uploading packs to XSOAR versions earlier than 6.5.0 is no longer supported."
+            "Use older versions of the Demisto-SDK for that (<=1.13.0)"
+        )
+    server_kwargs = {"skip_verify": "true"}
+
+    if skip_validations and target_demisto_version >= Version("6.6.0"):
```
[the remark this answers is not in the document]
Comment (dorschw) [reply demisto_sdk/commands/content_graph/objects/pack.py:96]: 6.5 != 6.6 :)

[the remark this answers is not in the document]
Comment (dantavori) [reply demisto_sdk/commands/content_graph/objects/pack.py:96]: I meant, put it in a constant

File: demisto_sdk/commands/content_graph/objects/classifier.py [hunk 3]
```diff
@@ -12,3 +13,7 @@ class Classifier(ContentItem, content_type=ContentType.CLASSIFIER):  # type: ign
 
     def metadata_fields(self) -> Set[str]:
         return {"name", "description"}
+
+    @classmethod
+    def _client_upload_method(cls, client: demisto_client) -> Callable:
```
[the remark this answers is not in the document]
Comment (dorschw) [reply demisto_sdk/commands/content_graph/objects/classifier.py:18]: Checked, can't do that, as the result is from a specific instance of the client :\

**Pull request merged**